client-sdk/ts-web/rt/src/types.ts

```diff
@@ -18,6 +18,43 @@ export interface CoreCallDataPublicKeyQueryResponse {
     public_key: KeyManagerSignedPublicKey;
 }
 
+/**
+ * Response to the RuntimeInfo query.
+ */
+export interface CoreRuntimeInfoQueryResponse {
+    runtime_version: runtime_info.Version;
+    state_version: number;
+    modules: {[key: string]: runtime_info.ModuleInfo};
+}
+
+export namespace runtime_info {
+    export interface Version {
+        major: number;
+        minor: number;
+        patch: number;
+    }
+
+    /**
+     * Metadata for an individual module within the runtime.
+     */
+    export interface ModuleInfo {
+        version: number;
+        params: any;
+        methods: MethodHandlerInfo[];
+    }
+
+    export interface MethodHandlerInfo {
+        name: string;
+        kind: MethodHandlerKind;
+    }
+
+    export enum MethodHandlerKind {
```

Comment:
Aspiring to autogenerate interfaces is an excellent reason for consistency. Not that consistency isn't quite excellent on its own :). Sorry, I did not spot the other enums. What this PR introduces is still a little fancier than `CallFormat` because I restrain the type to only the valid values, rather than any `string`. But if we ever autogenerate `string` as a type, that's going to be backwards compatible.

I believe the same is true for the Go interface: What this PR adds is a stricter with types than the `CallFormat` Go interface, but were we to autogenerate the new interface in the style of `CallFormat`, usage wouldn't change.

I now also got rid of the TS namespace.

Reply:
thanks!